packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -222,6 +222,9 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
     return result;
   }
 
+  @override
+  bool ignorePointerWhenCurrentDuringTransitions = true;
+
```

Comment:
Do we care if these continue to follow the `get ... => ...`  syntax that's defined in `ModalRoute`?

Reply:
yes, this also exposes a setter. Overrides only the getter sounds better.

Reply:
Since this is cupertino only thing, can this be implemented in cupertino library only? I imagine there is a lot of places we can insert ignore pointer

Reply:
So, I'm having trouble imagining how that would work. We'd need the logic to be shared between `MaterialPageRoute` (which I'm planning on updating as part of resolving #95764), `CupertinoPageRoute`, `_PageBasedCupertinoPageRoute`, `CupertinoModalPopupRoute` and `CupertinoDialogRoute`, and currently `ModalRoute` is the only class ancestor where that logic could be shared. I guess you could make a mixin that owns that logic, but even then, I think you'd need to add a new IgnorePointer on every route implementation. And further, if we pushed the `IgnorePointer` into the implementation of each of those routes, then I think they'd end up easily overriden by accident when people provide their own `pageBuilder`s or `transitionBuilder`s, to customize the transition behavior? I think this may be the only way to have this logic configurable for users simply by flipping a boolean value and consistently use the default behavior, which I think is an important feature to keep?

Reply:
But happy to continue to explore that if you think it would be preferable! Curious for your thoughts 👍🏻

Reply:
I realized you also need this for modal barrier as well, I think it makes sense to be in the modalroute